common/middleware/middleware.go

```
@@ -0,0 +1,214 @@
+// RequestIDMiddleware adds a unique request ID to each request
+func RequestIDMiddleware() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		requestID := c.Get("X-Request-ID")
+		if requestID == "" {
+			requestID = uuid.New().String()
+			c.Set("X-Request-ID", requestID)
+		}
+
+		// Store in both locals for middleware and context for logging
+		c.Locals("requestID", requestID)
+		ctx := context.WithValue(c.UserContext(), "requestID", requestID)
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)